### PR TITLE
Fix/pci rancher fix value tapc 1615

### PR DESCRIPTION
**Pull request by ppprevost**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop` 
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | Fix TAPC-1615
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [ ] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [] Breaking change is mentioned in relevant commits

### CODE REVIEW

File: packages/manager/apps/pci-rancher/src/components/layout-helpers/Dashboard/RancherDetail/RancherDetail.component.tsx [hunk 1]
```diff
@@ -89,11 +90,28 @@ const RancherDetail = ({
...
   const onAccessRancherUrl = () =>
     trackAction(TrackingPageView.DetailRancher, TrackingEvent.accessUi);
-
+  const { i18n } = useTranslation('common');
   const shouldDisplayUpdateSoftware =
     getLatestVersionAvailable(rancher, versions) &&
     isReadyStatus &&
     !updateSoftwareResponseType;
+  const locales = useRef({ ...dateFnsLocales }).current;
+  const userLocale = getDateFnsLocale(i18n.language);
```
Comment (SimonChaumet) [inline packages/manager/apps/pci-rancher/src/components/layout-helpers/Dashboard/RancherDetail/RancherDetail.component.tsx:99]: maybe this should be in a useMemo ?

File: packages/manager/apps/pci-rancher/src/components/layout-helpers/Dashboard/RancherDetail/RancherDetail.component.tsx [hunk 2]
```diff
@@ -89,11 +90,28 @@ const RancherDetail = ({
...
+
+  const displayDate = useCallback(
+    (value: string) => {
+      if (userLocale in locales) {
+        const localeId = userLocale as keyof typeof locales;
+        return format(new Date(dateUsage), value, {
+          locale: locales[localeId],
```
Comment (SimonChaumet) [inline packages/manager/apps/pci-rancher/src/components/layout-helpers/Dashboard/RancherDetail/RancherDetail.component.tsx:106]: Isn't date fns supposed to provide a default format ?

File: packages/manager/apps/pci-rancher/src/components/layout-helpers/Dashboard/RancherDetail/RancherDetail.component.tsx (hunk 1 above)
Comment (ppprevost) [reply packages/manager/apps/pci-rancher/src/components/layout-helpers/Dashboard/RancherDetail/RancherDetail.component.tsx:102]: We do not need to memoize it as it is a simple string value.

File: packages/manager/apps/pci-rancher/src/components/layout-helpers/Dashboard/RancherDetail/RancherDetail.component.tsx (hunk 2 above)
Comment (ppprevost) [reply packages/manager/apps/pci-rancher/src/components/layout-helpers/Dashboard/RancherDetail/RancherDetail.component.tsx:109]: I'm not entirely sure to understand ? What kind of default format ?

File: packages/manager/apps/pci-rancher/src/components/layout-helpers/Dashboard/RancherDetail/RancherDetail.component.tsx [hunk 3]
```diff
@@ -89,11 +90,28 @@ const RancherDetail = ({
 
   const onAccessRancherUrl = () =>
     trackAction(TrackingPageView.DetailRancher, TrackingEvent.accessUi);
-
+  const { i18n } = useTranslation('common');
```
[the remark this answers is not in the document]
Comment (ppprevost) [reply packages/manager/apps/pci-rancher/src/components/layout-helpers/Dashboard/RancherDetail/RancherDetail.component.tsx]: Right, fixed.

**Review by Eric-ciccotti — APPROVED** [review]

File: packages/manager/apps/pci-rancher/src/components/layout-helpers/Dashboard/RancherDetail/RancherDetail.component.tsx [hunk 4]
```diff
@@ -119,11 +125,27 @@ const RancherDetail = ({
...
+        return format(new Date(dateUsage), value, {
+          locale: locales[localeId],
+        });
+      }
+      return format(new Date(dateUsage), value, {
+        locale: locales.fr,
+      });
+    },
+    [userLocale, locales, dateUsage],
+  );
```
Comment (fredericvilcot) [inline packages/manager/apps/pci-rancher/src/components/layout-helpers/Dashboard/RancherDetail/RancherDetail.component.tsx:148]: Maybe we could simplify a bit :)
```suggestion
  const displayDate = useCallback(
    (value: string) =>
      format(new Date(dateUsage), value, {
        locale:
          userLocale in locales
            ? locales[userLocale as keyof typeof locales]
            : locales.fr,
      }),
    [userLocale, locales, dateUsage],
  );
```

**Review by fredericvilcot — APPROVED** [review]
Great!

**Review by Eric-ciccotti — APPROVED** [review]

File: packages/manager/apps/pci-rancher/src/components/Table/NumberCell/NumberCell.component.tsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+interface DisplayCellInterface {
+  cell: Cell<RancherService, number>;
+}
+
+function DisplayCellNumber({ cell }: Readonly<DisplayCellInterface>) {
+  const number = cell.renderValue();
+  return <OsdsText color={ODS_THEME_COLOR_INTENT.text}>{number}</OsdsText>;
+}
+
+export default DisplayCellNumber;
```
Comment (fredericvilcot) [inline packages/manager/apps/pci-rancher/src/components/Table/NumberCell/NumberCell.component.tsx:17]: I think you should name your component the same as your file (or the opposite)

Comment (ppprevost) [reply packages/manager/apps/pci-rancher/src/components/Table/NumberCell/NumberCell.component.tsx:17]: i was inspired by a Text Component and for consistency reason i will not change the name of this file.

**Review by fredericvilcot — APPROVED** [review]

**Review by Eric-ciccotti — DISMISSED** [review]

File: packages/manager/apps/pci-rancher/src/components/layout-helpers/Dashboard/RancherDetail/RancherDetail.component.tsx [hunk 6]
```diff
@@ -119,11 +125,23 @@ const RancherDetail = ({
 
   const onAccessRancherUrl = () =>
     trackAction(TrackingPageView.DetailRancher, TrackingEvent.accessUi);
-
   const shouldDisplayUpdateSoftware =
     getLatestVersionAvailable(rancher, versions) &&
     isReadyStatus &&
     !updateSoftwareResponseType;
+  const locales = useRef({ ...dateFnsLocales }).current;
```
Comment (frenautvh) [inline packages/manager/apps/pci-rancher/src/components/layout-helpers/Dashboard/RancherDetail/RancherDetail.component.tsx:132]: why using a ref here ? couldn't you use `dateFnsLocales` directly ?

**Review by frenautvh — APPROVED** [review]

**Review by frenautvh — APPROVED** [review]

**Review by fredericvilcot — APPROVED** [review]

**Review by fredericvilcot — APPROVED** [review]

**Review by anooparveti — APPROVED** [review]

**Pull request merged**